src/proposer/proposer.h

```diff
@@ -38,6 +42,10 @@ class Proposer {
 
   virtual bool IsStarted() = 0;
 
+  virtual std::shared_ptr<const CBlock> GenerateBlock(CWallet &wallet,
+                                                      const staking::CoinSet &coins,
+                                                      const boost::optional<CScript> &coinbase_script) = 0;
```

Comment:
Is this ever `none`?

Reply:
I see `ActiveProposerImpl` is passing `boost::none`. Is there a reason that passive and active proposer create block differently?

Reply:
Yes the active is used for proposing when the node runs with the proposing flag. The other is to be used when is instead called via rpc.

Reply:
But if I start the node with `-proposing=1` and will invoke manually the `generatetoaddress` RPC, the block will be created differently than if I have `-proposing=0`, right? 

I can see that we can have:
A. proposing itself and RPC command work differently
B. we disable RPC if node is proposing

But currently, it looks a bit inconsistent that the RPC will work differently depending on the `-proposing=` flag. I'd expect that it has always the same behavior.

Let me, please, know if I understood correctly and what you think about this.

Reply:
Yeah, I agree, I will check that in order to use the RPC the flag is set to `proposing=0`.